### ISSUE TITLE
Allow BASE_PATH to be set using en environment variable

### ISSUE THREAD

**Issue by jmfederico** [silverstripe/silverstripe-framework#8366]
## Affected Version

"4.2.1"

## Description

Currently there is no reliable way to set BASE_PATH when for some reason SilverStripe is not able to detect it automatically. This might happen for instance when COMPOSER_VENDOR_DIR is not "vendor" (see #8077).

Setting it in index.php is a partial solution, but it won't work for CLI usage (sake).

A solution would be to allow it to be set via a Environment Variable, the same way BASE_URL is being set.

Is this something you would approve? If so, I could work on a PR.

**Comment by tractorcow** [comment]
I disagree with allowing it to be configured. It should be auto-detected. Having to manually set it demonstrates the code is wrong if it doesn't detect it correctly. convention > configuration.
